wire-ios-request-strategy/Sources/Request Strategies/Federation/FederationTerminationManagerInterface.swift

```diff
@@ -0,0 +1,33 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public protocol FederationTerminationManagerInterface {
+
+    func handleFederationTerminationWith(_ domain: String)
+    func handleFederationTerminationBetween(_ domains: [String])
```

Comment:
I discussed handling this event with Kuba from android team and he said they log error when array has more than 2 domains. I think we should do the same because federating is one-to-one relation not many-to-many. Also it will be a lot of database requests at one time